### PR TITLE
eth: move eth.Config to a common package

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR moves the eth config definition into a separate/common package. So that both ETh/LES protocol can only import this common package.

### CODE REVIEW

**Review by zsfelfoldi — APPROVED** [review]

**Comment by zsfelfoldi** [comment]
Please fix the linter issue.

File: eth/ethconfig/config.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2017 The go-ethereum Authors
+// Copyright 2021 The go-ethereum Authors
```
Comment (gballet) [inline eth/ethconfig/config.go:1]: is it ok to change the copyright just because we move a file?

**Review by gballet — COMMENTED** [review]
I'm not a fan, I have a couple outstanding Prs that I will have to rebase _again_ because of this. Apart from that, LGTM.

File: eth/ethconfig/config.go (hunk 1 above)
Comment (rjl493456442) [reply eth/ethconfig/config.go:1]: not sure. I can change it back.

**Comment by fjl** [comment]
With this change, les still imports eth for `CreateConsensusEngine`. Maybe it should be moved to `ethconfig` as well.

**Comment by rjl493456442** [comment]
@fjl I can also make the change here. I think the `CreateConsensusEngine` probably can be moved to the `consensus` pkg. Let me check if it's fessible.

**Comment by rjl493456442** [comment]
@fjl I think all the dependency from the les to eth has been resolved. Please take another look.

**Review by fjl — APPROVED** [review]

**Pull request merged**